lib/mapUtils.js

```diff
@@ -43,3 +43,156 @@ export async function getProductData(store) {
+      const days = timeAndDaySplit[1].split('-');
+
+      // Case: Time corresponds to one day
+      if (days.length === 1) {
+        hoursPerDayDict[daysOfTheWeek[days[0]]] = time;
+      } else if (days.length === 2) {
+        const firstDay = days[0];
+        const lastDay = days[1];
+
+        // ['Sun', 'Mon', 'Tues', ...]
+        const indexFirstDay = daysOfTheWeekAbbrev.indexOf(firstDay);
+        const indexLastDay = daysOfTheWeekAbbrev.indexOf(lastDay);
+
+        let numDaysInInterval;
+
+        if (indexLastDay > indexFirstDay) {
+          // Case: Interval is Sun-Thurs, 0-4
+          numDaysInInterval = indexLastDay - indexFirstDay + 1;
+        } else {
+          // Case: Interval is Thurs-Sun, 4-0
+          numDaysInInterval = indexLastDay - indexFirstDay + 8;
+        }
+
+        [...Array(numDaysInInterval).keys()].map((i, _) => {
+          const dayIndex = (indexFirstDay + i) % 7;
+          hoursPerDayDict[daysOfTheWeekFull[dayIndex]] = time;
+        });
+      } else {
+        // Shouldn't ever have a case with more than two days in an interval.
+        console.error(
+          '[mapUtils] constructHoursDict: Incorrect store hours formatting in Airtable.'
+        );
+      }
+    });
+  }
+  return hoursPerDayDict;
+}
+
+export function computeStoreOpen(hours) {
+  if (hours === 'Open 24/7' || hours === 'Store hours unavailable') {
+    return hours;
+  }
+
+  const daysOfTheWeek = {
+    Sun: 'Sunday',
+    Mon: 'Monday',
+    Tu: 'Tuesday',
+    Wed: 'Wednesday',
+    Thurs: 'Thursday',
+    Fri: 'Friday',
+    Sat: 'Saturday',
+  };
+  const daysOfTheWeekFull = Object.values(daysOfTheWeek);
+  const today = new Date();
+  // currentTime is minutes since 12am
+  const currentTime = today.getHours() * 60 + today.getMinutes();
+
+  const todaysDayIndex = today.getDay();
+  const todaysDay = daysOfTheWeekFull[todaysDayIndex];
+
+  const hoursPerDayDict = constructHoursDict(hours);
+
+  // e.g. "8am-8pm"
+  const todaysStoreHours = hoursPerDayDict[todaysDay];
+  const hoursSplit = todaysStoreHours.split('-');
+  const openingTime = hoursSplit[0]; // e.g. "8:30am"
+  const closingTime = hoursSplit[1]; // e.g. "8:30pm"
+
+  const openingSuffix = openingTime.slice(-2);
+  const openingHourMin = openingTime.slice(0, -2);
+  const openingHourMinSplit = openingHourMin.split(':');
+  let openingHour = openingHourMinSplit[0];
+  const openingMin = openingHourMinSplit[1];
+
+  const closingSuffix = closingTime.slice(-2);
+  const closingHourMin = closingTime.slice(0, -2);
+  const closingHourMinSplit = closingHourMin.split(':'); // e.g. ["8", "30"]
+  let closingHour = closingHourMinSplit[0];
+  const closingMin = closingHourMinSplit[1];
+
+  // Convert to 24 hour time
+  if (openingSuffix === 'pm') {
+    openingHour += 12;
+  }
+
+  if (closingSuffix === 'pm') {
+    closingHour += 12;
+  }
+
+  const convertedOpeningTime = openingHour * 60 + openingMin;
+  const convertedClosingTime = closingHour * 60 + closingMin;
```

Comment:
Ah shoot I forgot the minutes were strings RIP thanks for catching that

Reply:
Ahh also the issue was that some of the times don't have minutes so I didn't check for minutes before converting something that didn't exist! So that's why it's NaN but that's fixed now tyty